packages/lit-element/src/lib/updating-element.ts

```diff
@@ -735,24 +743,6 @@ export abstract class UpdatingElement extends HTMLElement {
     }
   }
 
-  /**
-   * Requests an update which is processed asynchronously. This should
-   * be called when an element should update based on some state not triggered
-   * by setting a property. In this case, pass no arguments. It should also be
-   * called when manually implementing a property setter. In this case, pass the
-   * property `name` and `oldValue` to ensure that any configured property
-   * options are honored. Returns the `updateComplete` Promise which is resolved
-   * when the update completes.
-   *
-   * @param name {PropertyKey} (optional) name of requesting property
-   * @param oldValue {any} (optional) old value of requesting property
-   * @returns {Promise} A Promise that is resolved when the update completes.
-   */
-  requestUpdate(name?: PropertyKey, oldValue?: unknown) {
-    this.requestUpdateInternal(name, oldValue);
-    return this.updateComplete;
```

Comment:
Just a thought, but since this is a pretty silent breaking change (i.e. awaiting requestUpdate won't error), we _could_ add a devmode-only thing where we add a return like e.g.
```js
if (DEVMODE)
  return { 
    then() { 
      throw new Error(`In LitElement v3.0+, requestUpdate should not be awaited; instead await updateComplete`);
    }
  }
}
```

Although I guess you do get a TS warning: `'await' has no effect on the type of this expression`.

Anyway, maybe worth a TODO? Or maybe we should start an issue listing DEVMODE warnings?

Reply:
This would be good for non-TS users, and for non-await users.

I wonder if throwing in then() or returning a rejected Promise is better?